File: php/testcode/benchmark_test_00160.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00160(BenchmarkRequest $req): BenchmarkResponse {
    $varname = $req->param('key');
    $value = $req->param('value');
    $$varname = $value;
    return BenchmarkResponse::ok("set $varname");
}

File: rust/testcode/benchmark_test_00160.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let entry = LogEntry {
        user: req.param("user"),
        action: req.param("action"),
    };
    write_log(&format!("user={} action={}", entry.user, entry.action));
    super::shared::BenchmarkResponse::ok("Done")
}

struct LogEntry { user: String, action: String }

fn write_log(msg: &str) {
    eprintln!("[LOG] {}", msg);
}
